lib/bucket_store/key_storage.rb

```diff
@@ -40,45 +122,32 @@ def filename
     # @example Download a key
     #   BucketStore.for("inmemory://bucket/file.xml").download
     def download
-      raise ArgumentError, "Key cannot be empty" if key.empty?
```

Comment:
Don't we still need this?

Reply:
Nope: when we call `stream` it does this check for both uploads / downloads